src/Microsoft.Extensions.DependencyInjection.Abstractions/ServiceDescriptor.cs

```diff
@@ -316,7 +211,7 @@ public static ServiceDescriptor Singleton<TService>(TService implementationInsta
             return Singleton(typeof(TService), implementationInstance);
         }
 
-        public static ServiceDescriptor Singleton(
+        public static InstanceServiceDescriptor Singleton(
```

Comment:
These seems to be missing their `EnumerableServiceDescriptor` and `OrderedEnumerableServiceDescriptor` buddies 😢 Is that intentional?